Gateway/Http/Client/TransactionEmailRefund.php

```diff
@@ -0,0 +1,150 @@
+     * @param RefundCallback     $refundCallback
+     */
+    public function __construct(
+        private readonly PaytrailLogger $log,
+        private readonly Adapter $paytrailAdapter,
+        private readonly EmailRefundRequest $emailRefundRequest,
+        private readonly RefundCallback $refundCallback
+    ) {
+    }
+
+    /**
+     * PlaceRequest function
+     *
+     * @param TransferInterface $transferObject
+     *
+     * @return array
+     */
+    public function placeRequest(TransferInterface $transferObject): array
+    {
+        $request       = $transferObject->getBody();
+
+        return $this->emailRefund(
+            $request['order'],
+            $request['amount'],
+            $request['parent_transaction_id']
+        );
+    }
+
+    /**
+     * EmailRefund function
+     *
+     * @param OrderAdapterInterface|null $order
+     * @param float|null                 $amount
+     * @param string|null                $transactionId
+     *
+     * @return array
+     */
+    public function emailRefund(
+        OrderAdapterInterface $order = null,
+        float $amount = null,
+        string $transactionId = null
+    ) {
+        $response= [];
+
+        try {
+            $paytrailClient = $this->paytrailAdapter->initPaytrailMerchantClient();
+
+            $this->log->debugLog(
+                'request',
+                \sprintf(
+                    'Creating %s request to Paytrail API %s',
+                    'email_refund',
+                    isset($order) ? 'With order id: ' . $order->getId() : ''
+                )
+            );
+
+            // Handle request
+            $this->setEmailRefundRequestData($this->emailRefundRequest, $amount, $order);
+
+            $emailRefundResponse = $paytrailClient->emailRefund($this->emailRefundRequest, $transactionId);
+            $response = $this->formatResponse($emailRefundResponse);
+
+            $this->log->debugLog(
+                'response',
+                sprintf(
+                    'Response for email_refund. Transaction Id: %s',
+                    $emailRefundResponse->getTransactionId()
+                )
+            );
+        } catch (RequestException $e) {
+            if ($e->hasResponse()) {
+                $this->log->error(\sprintf(
+                    'Connection error to Paytrail Payment Service API: %s Error Code: %s',
+                    $e->getMessage(),
+                    $e->getCode()
+                ));
+                $response["error"]  = $e->getMessage();
+            }
+        } catch (\Exception $e) {
+            $this->log->error(
+                \sprintf(
+                    'A problem occurred during Paytrail Api connection: %s',
+                    $e->getMessage()
+                ),
+                $e->getTrace()
+            );
+            $response["error"]  = $e->getMessage();
+        }
+
+        return $response;
+    }
+
+    /**
+     * CreateRefundCallback function
```

Comment:
Mistake in function name
Maybe move this function to separate class under Model/Refund/EmailRefundDataProvider.php?